index.js

```diff
@@ -44,19 +44,52 @@ function timeout (time, options) {
   var respond = opts.respond === undefined || opts.respond === true
 
   return function (req, res, next) {
-    var id = setTimeout(function () {
-      req.timedout = true
-      req.emit('timeout', delay)
-    }, delay)
+    var started = Date.now()
+    var id = createTimeout(req, delay)
 
     if (respond) {
-      req.on('timeout', onTimeout(delay, next))
+      req.on('timeout', function () {
+        next(createError(503, 'Response timeout', {
+          code: 'ETIMEDOUT',
+          timeout: delay
+        }))
+      })
     }
 
     req.clearTimeout = function () {
+      delay = 0
       clearTimeout(id)
     }
 
+    req.resetTimeout = function (newDelay) {
```

Comment:
If calling reset after the timeout already occurred, should req.timedout be reset back to false? Or should is stay true and update docs to describe it just represents the first trigger? Or not allow a reset after timeout happened?

Reply:
it should probably remain false to let the user know that it was triggered.

Reply:
It remains `true` if `req.clearTimeout()` is called after timeout is fired.

I updated the docs to reflect this.